### PR TITLE
XRRDEV-2196 API to add a configuration signing key

### PR DESCRIPTION
**Pull request by VPaliliunas**
https://nordic-institute.atlassian.net/browse/XRDDEV-2196

[XRDDEV-2196]: https://nordic-institute.atlassian.net/browse/XRDDEV-2196?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/centralserver/admin-service/core/src/main/java/org/niis/xroad/centralserver/restapi/service/ConfigurationSigningKeyServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import java.util.UUID;
+
+import static org.niis.xroad.centralserver.restapi.service.exception.ErrorMessage.CONFIGURATION_NOT_FOUND;
+import static org.niis.xroad.centralserver.restapi.service.exception.ErrorMessage.KEY_GENERATION_FAILED;
+import static org.niis.xroad.cs.admin.api.dto.PossibleAction.GENERATE_KEY;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class ConfigurationSigningKeyServiceImpl implements ConfigurationSigningKeyService {
```
Comment (andresrosenthal) [inline src/centralserver/admin-service/core/src/main/java/org/niis/xroad/centralserver/restapi/service/ConfigurationSigningKeyServiceImpl.java:62]: We're pretty much stepping on each other's toes with the given epic, so as a result it has pretty much become a race condition, but the given implementation and it's interface should be unified with the existing [ConfigurationSigningKeysServiceImpl](https://github.com/nordic-institute/X-Road/blob/develop-7.x/src/centralserver/admin-service/core/src/main/java/org/niis/xroad/centralserver/restapi/service/ConfigurationSigningKeysServiceImpl.java)

File: src/centralserver/admin-service/core-api/src/main/java/org/niis/xroad/cs/admin/api/dto/ConfigurationSigningKey.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import lombok.experimental.Accessors;
+
+import java.time.Instant;
+import java.util.List;
+
+@Data
+@Accessors(chain = true)
+@AllArgsConstructor
+@NoArgsConstructor
+public class ConfigurationSigningKey {
```
Comment (andresrosenthal) [inline src/centralserver/admin-service/core-api/src/main/java/org/niis/xroad/cs/admin/api/dto/ConfigurationSigningKey.java:41]: Although different regarding its properties, this domain object class is already existent in the [org.niis.xroad.cs.admin.api.domain](https://github.com/nordic-institute/X-Road/tree/develop-7.x/src/centralserver/admin-service/core-api/src/main/java/org/niis/xroad/cs/admin/api/domain) package.
As they serve the same purpose & there is a plan to unify these packages into a single target, it'd be better to adjust early.

File: src/centralserver/admin-service/core/src/main/java/org/niis/xroad/centralserver/restapi/service/ConfigurationSigningKeyServiceImpl.java (hunk 1 above)
Comment (ovidijusnortal) [reply src/centralserver/admin-service/core/src/main/java/org/niis/xroad/centralserver/restapi/service/ConfigurationSigningKeyServiceImpl.java]: Refactored.

File: src/centralserver/admin-service/core-api/src/main/java/org/niis/xroad/centralserver/restapi/service/exception/ErrorMessage.java [hunk 3]
```diff
@@ -81,10 +81,14 @@ public enum ErrorMessage {
     SIGNER_PROXY_ERROR("Signer proxy exception"),
     TOKEN_INCORRECT_PIN_FORMAT("Incorrect PIN format"),
     TOKEN_ACTION_NOT_POSSIBLE("Token action not possible"),
+    TOKEN_FULL("2 keys already exist on this token"),
+    KEY_GENERATION_FAILED("Singing key generation failed"),
+    KEY_DELETE_FAILED("Singing key delete failed"),
```
Comment (ovidijusnortal) [inline src/centralserver/admin-service/core-api/src/main/java/org/niis/xroad/centralserver/restapi/service/exception/ErrorMessage.java:86]: are enums: TOKEN_FULL and KEY_DELETE_FAILED used somewhere?

**Pull request merged**